connectors/vault/vaultconn-bl/vault_reader.go

```diff
@@ -0,0 +1,33 @@
+// Copyright 2020 IBM Corp.
+// SPDX-License-Identifier: Apache-2.0
+
+package vaultconnbl
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	vaultutils "github.com/ibm/the-mesh-for-data/connectors/vault/vault_utils"
+	pb "github.com/ibm/the-mesh-for-data/pkg/connectors/protobuf"
+)
+
+type Server struct {
+}
+
+func (s *Server) GetCredentialsInfo(in *pb.DatasetCredentialsRequest, vault vaultutils.VaultConnection, vaultPathKey string) (*pb.DatasetCredentials, error) {
+	log.Println("Vault connector: GetCredentialsInfo")
+	log.Printf("Vault connector: GetCredentialsInfo, in = %v\n", in)
+
+	// secretaddr := url.QueryEscape(in.DatasetId)
+	secretaddr := strings.ReplaceAll(in.DatasetId, " ", "")
+
+	readCredentials, err := vault.GetFromVault(vaultPathKey, secretaddr)
```

Comment:
As far as I can see only `Client` from `VaultConnection` is used. The method `GetFromVault` can be rewritten to only use `api.Client` as input instead the whole `VaultConnection` that it does not use. That would also make the test easier as `VaultConfig` and `VaultConnection` are not needed.

Reply:
Though only `Client ` from `VaultConnection` is being used in the `GetFromVault` function, the `vault` object in Line 25 is being reused after [it was created](https://github.com/rohithdv/the-mesh-for-data/blob/255e89d8e98483b9261bbd3f54186f2259d74088/connectors/vault/vault_connector.go#L60) as part of initialisation when the vault connector started. During this initialisation, the `VaultConfig` object is being used. We are trying to reuse a single object for different purposes so that the entire functionality can be linked to a single `VaultConnection` object. Please let me know if this is ok. Thanks. @froesef

Reply:
If you move the initialization into the `CreateVaultConnection` you don't even need a `VaultConnection` struct because you're only working on `api.Client` afterwards. Please correct me if I'm getting this wrong. But then the `VaultConnection` struct is only used to organize and group the GetFromVault and AddToVault methods under one structure. 
Instead of `func (vlt *VaultConnection) GetFromVault(vaultPathKey string, innerVaultPath string) (string, error)` you could have `func GetFromVault(client *api.Client, vaultPathKey string, innerVaultPath string) (string, error)`

Reply:
Actually at a high level `VaultConnection` is abstracting out the vault related functions (which are defined in vault_utils.go). If we dont have `VaultConnection` struct, then 
(1) there will be changes to the `InitVault() ` . we will need to pass Config object to it as a parameter.
(2) Also there will be changes to` GetFromVault() `and `AddToVault()`  as you have mentioned. 
(3) Also we need to import vault related libraries in vault_connector.go (`"github.com/hashicorp/vault/api"`) which was earlier imported in vault_utils.go.
(4)` CreateVaultConnection()` should return` *api.Client` instead of `VaultConnection`. 
 Should we go ahead in this direction ? Please advise. Thanks. @froesef

Reply:
I think in the end it's a question of programming style. I would do it the way you described earlier. But if you want to keep `VaultConnection` in order to keep everything encapsulated behind it that's also fine. I think we're all quite new to Go.

Reply:
I agree with your view. Since it is mainly a matter of programming style I think it would be ok to keep the design as it is now if it is fine with you. This design had emerged due to discussion with other governance team members. Unless there is a real need to change the design I would prefer to keep the design unchanged. Please let me know if it is ok. Thanks.